plugins/de.cognicrypt.core/src/de/cognicrypt/core/Constants.java

```diff
@@ -555,11 +556,28 @@ public String toString() {
 	// define a correct ID (get range of possible ones)
 	public static final int JDT_PROBLEM_ID = 10000000;
 	public final static String RELATIVE_RULES_DIR = "resources/CrySLRules";
+	public final static String ECLIPSE_RULES_DIR = System.getProperty("user.dir");
 	public static final String cryslFileEnding = ".cryptsl";
 	public static final String cryslEditorID = "de.darmstadt.tu.crossing.CryptSL";
 	public static final String HEALTHY = "Secure";
 	public static final String UNHEALTHY = "Insecure";
 	
 	// define the max java version before which plugin works.
 	public static final String CC_JAVA_VERSION = "1.8";
+	
+	//Preference page rules table Constants
+	public static final String TABLE_HEADER_RULES = "Rules";
+	public static final String TABLE_HEADER_VERSION = "Version";
+	public static final String TABLE_HEADER_URL = "URL";
+	
+	public static final String JCA_TABLE_ITEM = "Java Cryptographic Architecture";
+	public static final String BC_TABLE_ITEM = "Bouncy Castle";
+	public static final String TINK_TABLE_ITEM = "Tink";
+	
+	public static final String NEXUS_SOOT_RELEASE = "https://soot-build.cs.uni-paderborn.de/nexus/repository/soot-release";

```

Comment:
Please extract all URLS to an external config file.

Reply:
Do you mean a separate java file which consists of only URLs?